### PR TITLE
Improving criteria table parsing in HRA

### LINKED ISSUE #1191: HRA criteria table parsing throws cryptic error when section header is missing
Looking at a [user's forums post](https://community.naturalcapitalproject.org/t/hra-model-unboundlocalerror-local-variable-current-stressor-header-row/3424/2), the following exception was being raised:

```
>>> hra._parse_criteria_table('/Users/jdouglass/Downloads/2023-02-18-manipreyamrakash-hra-criteria-table/criteria_score.csv', 'target.csv')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/jdouglass/workspace/phargogh/invest/src/natcap/invest/hra.py", line 1781, in _parse_criteria_table
    attribute_name = table[current_stressor_header_row][col_index]
UnboundLocalError: local variable 'current_stressor_header_row' referenced before assignment
```

The problem is that the user's criteria table has the field `"HABITAT RESILIENCE ATIRIBUTES "` instead of `"HABITAT RESILIENCE ATTRIBUTES"`.  The typo is causing the program to break in a cryptic way.

### PR DESCRIPTION
**Pull request by phargogh**
This PR improves the usability of HRA's criteria table parsing by:

1. stripping all leading/trailing whitespace from string table cells (which should generally improve the likelihood of string matches between fields that must match)
2. explicitly checking that certain required section headers are found in the table.

The user's reported exception was due to number 2 above.  Number 1 became clear that it might be useful by looking at the user's table.

This PR depends on #1190 .

Fixes #1191 

## Checklist
- [x] Updated HISTORY.rst and link to any relevant issue (if these changes are user-facing)
- [ ] Updated the user's guide (if needed)
- [ ] Tested the affected models' UIs (if relevant)

### CODE REVIEW

**Comment by davemfish** [comment]
@phargogh I've seen this review request but was waiting until #1190 was merged so that those changes no longer appear here. Let me know if that's not a good idea for some reason.

**Comment by phargogh** [comment]
I think that's fine @davemfish , thanks!

File: src/natcap/invest/hra.py
Comment (davemfish) [inline src/natcap/invest/hra.py:1725]: Does it make sense to print `missing_sections` instead of `required_section_headers` in this message? Or else maybe rephrase to "The criteria table requires the following section headers:"

**Review by davemfish — COMMENTED** [review]
@phargogh , looks good. I have a minor comment about content of the error message.

And just to confirm, do you still intend these Scenic Quality changes to be included? They look good as well.

File: src/natcap/invest/hra.py
Comment (phargogh) [reply src/natcap/invest/hra.py]: Whoops, thanks for catching this!  I've patched it in 7fff675

**Review by davemfish — APPROVED** [review]

**Pull request merged**
